sunkit_instruments/abstractions/channels.py

```diff
@@ -0,0 +1,29 @@
+"""This module defines abstractions related to instrument channels."""
+
+__all__ = ["AbstractChannel"]
+
+import abc
+
+import astropy.units as u
+
+
+class AbstractChannel(abc.ABC):
+    """An abstract base class for defining instrument channels.
+
+    .. caution::
+
+       This abstract class is still under development and may change
+       in the near future.
+    """
+
+    @abc.abstractmethod
+    def effective_area(self) -> u.cm**2:
+        ...
+
+    @abc.abstractmethod
+    def gain(self) -> u.DN / u.photon:
+        ...
```

Comment:
```suggestion
    def gain(self) -> u.DN / u.photon:
        return 1 * u.DN / u.photon
```
If we default to one, we'd want to remove the `@abstractmethod` decorator.

I'm not sure if we'd want this to be dimensional or dimensionless, though.

Reply:
The gain typically provides a conversion from electrons to counts (or DN) so I think it should have dimensions.